AutoPBW.WPF/MainWindow.xaml.cs

```diff
@@ -155,167 +155,175 @@ private void RefreshData()
+								currentTurnProcess.StartInfo = game.ProcessTurnPrepare();
+								currentTurnProcess.EnableRaisingEvents = true;
+								currentTurnProcess.Exited += process_Exited;
+								btnReset.Foreground = Brushes.Red;
+								currentTurnProcess.Start();
+								break; // wait till we finish this one
+							}
+							catch (Exception ex)
+							{
+								ShowBalloonTip("Turn processing failed", "Turn processing for " + game + " failed:\n" + ex.Message + "\n" + ex.Message.Split('\n').First(), game);
+								game.PlaceHold(ex.Message);
+								if (currentTurnProcess != null)
+								{
+									currentTurnProcess.Close();
+									currentTurnProcess = null;
+									HostGame.ProcessingGame = null;
+									currentTurnExitCode = null;
+									HostGame.ProcessingGame = null;
+									btnReset.ClearValue(Control.ForegroundProperty);
+								}
+								gamesToProcess.Remove(game);
 								HostGame.ProcessingGame = null;
-								btnReset.ClearValue(Control.ForegroundProperty);
 							}
-							gamesToProcess.Remove(game);
-							HostGame.ProcessingGame = null;
 						}
-					}
-					else
-					{
-						PBW.Log.Write($"Can't process {game}: unconfigured mod {game.Mod} or engine {game.Engine}.");
-						gamesToProcess.Remove(game); // can't process this game now
+						else
+						{
+							PBW.Log.Write($"Can't process {game}: unconfigured mod {game.Mod} or engine {game.Engine}.");
+							gamesToProcess.Remove(game); // can't process this game now
+						}
 					}
 				}
-			}
 
-			// auto-download
-			if (Config.Instance.AutoDownload && playerGames != null)
-			{
-				var waitingPLR = playerGames.Where(g => g.Status == PlayerStatus.Waiting && !g.HasDownloaded && g.PlayerNumber > 0 && g.TurnNumber > 0);
-				foreach (var ng in waitingPLR)
+				// auto-download
+				if (Config.Instance.AutoDownload && playerGames != null)
 				{
-					try
+					var waitingPLR = playerGames.Where(g => g.Status == PlayerStatus.Waiting && !g.HasDownloaded && g.PlayerNumber > 0 && g.TurnNumber > 0);
+					foreach (var ng in waitingPLR)
 					{
-						ng.DownloadTurn();
-					}
-					catch (Exception ex)
-					{
-						ShowBalloonTip("Download failed", "Auto-download for " + ng + " failed:\n" + ex.Message + "\n" + ex.Message.Split('\n').First(), ng);
+						try
+						{
+							ng.DownloadTurn();
+						}
+						catch (Exception ex)
+						{
+							ShowBalloonTip("Download failed", "Auto-download for " + ng + " failed:\n" + ex.Message + "\n" + ex.Message.Split('\n').First(), ng);
+						}
 					}
 				}
-			}
 
-			// remember selection
-			var newGame = gridPlayerGames.Items.Cast<PlayerGame>().SingleOrDefault(g => g.Code == selCode && g.PlayerNumber == selPlayerNumber);
-			gridPlayerGames.SelectedItem = newGame;
+				// remember selection
+				var newGame = gridPlayerGames.Items.Cast<PlayerGame>().SingleOrDefault(g => g.Code == selCode && g.PlayerNumber == selPlayerNumber);
+				gridPlayerGames.SelectedItem = newGame;
+			}

```

Comment:
This is the new closing brace - everything between here and my other comment is just indented one tab - but that drives github crazy